ci/tasks/check-upstream-status.py

```diff
@@ -0,0 +1,66 @@
+""" For a commit that triggered the job, verify that all the checks successfully
+finished before trying to merge the upstream changes. Upstream checks can take up
+to 30 minutes, so we check multiple times (max 10), and wait 5 minutes (300 sec) 
+between the attempts.
+"""
+
+import argparse
+import logging
+import sys
+import time
+
+import requests
+
+logger = logging.getLogger("check-upstream-status")
+logging.basicConfig(level=logging.INFO)
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--commit", required=True, help="Upstream commit hash")
+# Github repository access token is used to avoid errors due to API rate limits
+parser.add_argument("--token", required=True, help="Github repository access token")
+args = parser.parse_args()
+
+num_check = 0
+max_checks = 10
+
+while num_check < max_checks:
+    jobs = []
+    job_status = []
+    status_filter = []
+    data = requests.get(
+        f"https://api.github.com/repos/dependabot/dependabot-core/commits/{args.commit}/check-runs",
+        headers={"authorization":f"Bearer {args.token}", "Accept": "application/vnd.github.antiope-preview+json"},
+    )
+    if data.status_code != requests.codes.ok:
+        num_check += 1
+        logger.error("HTTP request failed ")
+        logger.info(f"Waiting before checking upstream status again. This was attempt number: {num_check}")
+        time.sleep(1)
+        continue
+
+    response = data.json()
+    job_status = [ (check_run["status"],check_run["conclusion"]) for check_run in response["check_runs"] ]
+    status_filter = list(filter(lambda item: item != ('completed', 'success') ,job_status))
+    status = any(status_filter)
+    
+    if not status:
```

Comment:
The following is easier to understand for me:
```suggestion
    all_ok = all(item == ('completed', 'success') for item in job_status)
    
    if all_ok:
```